exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -723,19 +723,20 @@ private void init(MetadataContext metaContext) throws IOException {
         // if querying a single file we can look up the metadata directly from the file
         metaPath = new Path(p, Metadata.METADATA_FILENAME);
       }
-      if (metaPath != null && fs.exists(metaPath)) {
+      if (metaPath != null && fs.exists(metaPath) && Metadata.MetadataVersion.isVersionSupported(fs, metaPath)) {
```

Comment:
Nit. No need to check `fs.exists` here as we, in effect, check this when trying to open the file. Handle the file not found exception instead.

Note that Drill is a multi-user system. Even with the `exists` check, there is still a race condition where the file can exist during the check, but be deleted by the time we try to open the file. So, since we have to check on open anyway, no need to check here.